src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SingleSpaceSeparatorCheck.java

```diff
@@ -193,8 +195,7 @@ private boolean isTextSeparatedCorrectlyFromPrevious(String line, int columnNo)
      *         not preceded by another space.
      */
     private static boolean isSingleSpace(String line, int columnNo) {
-        return !isPrecededByMultipleWhitespaces(line, columnNo)
-                && isSpace(line, columnNo);
```

Comment:
One condition inside `isPrecededBy`  and `isSpace` were doing the same check so I moved the surviving condition out and placed it in this method.